clang/test/AST/ast-attr-add-ir-annotations-pack.cpp

```diff
@@ -0,0 +1,848 @@
+// RUN: not %clang_cc1 -fsycl-is-device -std=gnu++11 -ast-dump %s | FileCheck %s
+
+// Tests the AST produced from instantiating templates using
+// __sycl_detail__::add_ir_annotations_member attributes with pack expansion
+// arguments.
+
+constexpr const char AttrName1[] = "Attr1";
+constexpr const char AttrName2[] = "Attr2";
+constexpr const char AttrName3[] = "Attr3";
+constexpr const char AttrVal1[] = "Val1";
+constexpr const char AttrVal2[] = "Val2";
+constexpr const char AttrVal3[] = "Val3";
+
+template <int... Is> struct ClassWithAnnotFieldTemplate1 {
```

Comment:
Does this test actually check the same thing as `clang/test/AST/ast-attr-add-ir-attributes-packs.cpp`?

Reply:
Yes, the `ast-attr-add-ir-annotations` tests are testing roughly the same cases as their `ast-attr-add-ir-attributes` counterparts. Same goes for the sema tests. The reason for the separation is the names, i.e. `add-ir-attributes` and `add-ir-annotations`.